### DISCUSSION TITLE
Interface to Neper or Dream3d for the generation of representative volume elements

### DISCUSSION THREAD

**Discussion by ralfHielscher** [mtex-toolbox/mtex#531] [Feature Requests]
Implement an MTEX interface to [Neper](https://neper.info/) or [Dream3d](https://github.com/BlueQuartzSoftware/DREAM3D) for the generation of an RVE (Representiv Volume Element)

Input:
* grains statistics (size, shape, number of neighbors, number of phases)
* texture (ODF)

Output: (needs to be added)
* 
*

**Comment by oMuransky** [comment]
Hi Ralf,

Thank you for your interest. It is a bit difficult to quickly explain, however, in short, RVE in this case refers to a synthetic microstructure, which would have characteristics of the measured EBSD map/s. The input would be the grain statistic (size, shape, number of neighbors, number of phases, etc), and texture (ODF). The output would be a synthetic microstructure, which would statistically represent EBSD map/s - typically a text file containing positions of voxels (or pixels), Euler angles, phases, volume fraction, etc. I think the best would be to follow input file for DAMASK crystal plasticity code: https://damask.mpie.de/bin/view/Examples/WebHome

I'll keep thinking about it. At the moment I'm using MTEX for EBSD analysis and the DREAM3D for RVE with a limited success. Let's see if anyone else is interested. 

Many thanks,
Ondrej

**Comment by junspark** [comment]
Hello,
I believe neper is a descent tool for this.
https://neper.info/

________________________________________

**Reply by oMuransky** [reply]
Thanks @junspark. I'm aware of NEPER, it is indeed a decent tool. I, however, feel that I'm still starting with MTEX and then taking the statistical data into another program. If we have all the information on the microstructure in MTEX after analyzing EBSD map/s it would be more straightforward process to generate representative microstructure directly in MTEX. In that case we might be able to incorporate also additional information on the pre-existing subgrains, GNDs and misorientation. In any case it is just an idea, which I thought would be useful.

**Comment by junspark** [comment]
I agree that unifying / streamlining workflow is a good thing in general. 
But a lot of these softwares (Neper / Dream3D) are specialized tools and might be quite expensive to implement in matlab.
Perhaps, one could consider calling Neper or Dream3D mesh generator from matlab / mtex? 
- jsp

________________________________________

**Reply by oMuransky** [reply]
That's a joy of Matlab, I guess everyone would now prefer if MTEX is written in Python. However, MTEX is already doing GND calculations, which are computationally expensive already so I don't think that calculating RVEs would be that much more computationally expensive. Maybe there can be a way to pass all the information on the microstructure from MTEX to DREAM3D but at present it is done manually and not all the information can be passed on - this results in somewhat limited RVE - at least that is my experience.
